gci/src/main/java/com/gcinterceptor/gci/HeapMonitor.java

```diff
@@ -0,0 +1,40 @@
+package com.gcinterceptor.gci;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+
+public class HeapMonitor {
+	private MemoryPoolMXBean youngPool;
+	private GarbageCollector collector;
+	private long lastAlloc; 
+	private long lastUsed;
+
+	public HeapMonitor() {
+		for (final MemoryPoolMXBean pool : ManagementFactory.getMemoryPoolMXBeans()) {
+			// TODO(danielfireman): Generalize this to other JVM versions.
```

Comment:
I know why this is here. Feel free to remove this TODO.

Reply:
Done